projects/igniteui-angular/src/lib/query-builder/query-builder-tree.component.ts

```diff
@@ -0,0 +1,1673 @@
+
+    private setFormat(field: FieldType) {
+        if (!field.pipeArgs) {
+            field.pipeArgs = { digitsInfo: DEFAULT_PIPE_DIGITS_INFO };
+        }
+
+        if (!field.pipeArgs.format) {
+            field.pipeArgs.format = field.dataType === DataType.Time ?
+                DEFAULT_PIPE_TIME_FORMAT : field.dataType === DataType.DateTime ?
+                    DEFAULT_PIPE_DATE_TIME_FORMAT : DEFAULT_PIPE_DATE_FORMAT;
+        }
+    }
+
+    private selectDefaultCondition() {
+        if (this.selectedField && this.selectedField.filters) {
+            this.selectedCondition = this.selectedField.filters.conditionList().indexOf('equals') >= 0 ? 'equals' : this.selectedField.filters.conditionList()[0];
+        }
+    }
+
+    private setFilters(field: FieldType) {
+        if (!field.filters) {
+            switch (field.dataType) {
+                case DataType.Boolean:
+                    field.filters = IgxBooleanFilteringOperand.instance();
+                    break;
+                case DataType.Number:
+                case DataType.Currency:
+                case DataType.Percent:
+                    field.filters = IgxNumberFilteringOperand.instance();
+                    break;
+                case DataType.Date:
+                    field.filters = IgxDateFilteringOperand.instance();
+                    break;
+                case DataType.Time:
+                    field.filters = IgxTimeFilteringOperand.instance();
+                    break;
+                case DataType.DateTime:
+                    field.filters = IgxDateTimeFilteringOperand.instance();
+                    break;
+                case DataType.String:
+                default:
+                    field.filters = IgxStringFilteringOperand.instance();
+                    break;
+            }
+        }
+    }
+
+
+    private addGroup(operator: FilteringLogic, parent?: ExpressionGroupItem, afterExpression?: ExpressionItem) {
+        this.cancelOperandAdd();
+
+        const groupItem = new ExpressionGroupItem(operator, parent);
+
+        if (parent) {
+            if (afterExpression) {
+                const index = parent.children.indexOf(afterExpression);
+                parent.children.splice(index + 1, 0, groupItem);
+            } else {
+                parent.children.push(groupItem);
+            }
+        } else {
+            this.rootGroup = groupItem;
+        }
+
+        this.addCondition(groupItem);
+        this.currentGroup = groupItem;
+    }
+
+    private createExpressionGroupItem(expressionTree: IExpressionTree, parent?: ExpressionGroupItem, entityName?: string): ExpressionGroupItem {
+        let groupItem: ExpressionGroupItem;
+        if (expressionTree) {
+            groupItem = new ExpressionGroupItem(expressionTree.operator, parent);
+            if (!expressionTree.filteringOperands) {
+                return groupItem;
+            }
+
+            for (const expr of expressionTree.filteringOperands) {
```

Comment:
for of loops look cool but they are a lot ot slower than traditional for i; i< array.length . Sometimes traditional is better :)